File: sdk/python/sylora_client.py
"""SYLORA Python SDK foundation for the Developer Platform sandbox."""

from __future__ import annotations

import json
import urllib.request
from typing import Any, Optional


class SyloraClient:
    def __init__(self, base_url: str = "http://localhost:8787", api_key: Optional[str] = None) -> None:
        self.base_url = base_url.rstrip("/")
        self.api_key = api_key

    def request(self, path: str, method: str = "GET", body: Optional[dict[str, Any]] = None) -> dict[str, Any]:
        data = None if body is None else json.dumps(body).encode("utf-8")
        req = urllib.request.Request(f"{self.base_url}{path}", data=data, method=method)
        req.add_header("content-type", "application/json")
        if self.api_key:
            req.add_header("authorization", f"Bearer {self.api_key}")
        with urllib.request.urlopen(req) as res:
            return json.loads(res.read().decode("utf-8"))

    def identity_me(self) -> dict[str, Any]:
        return self.request("/api/v1/identity/me")

    def list_agents(self) -> dict[str, Any]:
        return self.request("/api/agents")

File: sdk/js/sylora-client.js
/** Minimal SYLORA JS/TS-ready client for Developer Platform sandbox keys. */
export class SyloraClient {
  constructor({ baseUrl = 'http://localhost:8787', apiKey } = {}) {
    this.baseUrl = String(baseUrl).replace(/\/$/, '');
    this.apiKey = apiKey;
  }

  async request(path, { method = 'GET', body } = {}) {
    const headers = { 'content-type': 'application/json' };
    if (this.apiKey) headers.authorization = `Bearer ${this.apiKey}`;
    const res = await fetch(`${this.baseUrl}${path}`, {
      method,
      headers,
      body: body == null ? undefined : JSON.stringify(body)
    });
    const data = await res.json().catch(() => ({}));
    if (!res.ok) throw Object.assign(new Error(data.error || 'SYLORA_API_ERROR'), { status: res.status, data });
    return data;
  }

  get identity() {
    return {
      me: () => this.request('/api/v1/identity/me')
    };
  }

  get agents() {
    return {
      list: () => this.request('/api/agents')
    };
  }

  get status() {
    return {
      ecosystem: () => this.request('/api/ecosystem/status')
    };
  }
}
